front/lib/api/assistant/conversation.ts

```diff
@@ -1655,3 +1669,39 @@ async function* streamRunAgentEvents(
     }
   }
 }
+
+async function isMessagesLimitReached({
+  owner,
+  plan,
+}: {
+  owner: WorkspaceType;
+  plan: PlanType;
+}): Promise<boolean> {
+  if (plan.limits.assistant.maxMessages === -1) {
+    return false;
+  }
+  /** Counting on workspace because user can be null */
+  const count = await Conversation.count({
+    include: [
+      {
+        model: Message,
+        required: true,
+        as: "messages",
+        include: [
```

Comment:
Let's remove that condition :+1:

Reply:
You mean the `as: messages`? I believe the rest is necessary to count all and only agentmessages